bot/exts/fun/madlibs.py

```diff
@@ -0,0 +1,116 @@
+import json
+from asyncio import TimeoutError
+from pathlib import Path
+from random import choice
+from typing import TypedDict
+
+import discord
+from discord.ext import commands
+
+from bot.bot import Bot
+from bot.constants import Colours
+
+TIMEOUT = 60.0
+
+
+class MadlibsTemplate(TypedDict):
+    """Structure of a template in the madlibs JSON file."""
+
+    title: str
+    blanks: list[str]
+    value: list[str]
+
+
+class Madlibs(commands.Cog):
+    """Cog for the Madlibs game."""
+
+    def __init__(self, bot: Bot):
+        self.bot = bot
+        self.templates = self._load_templates()
+
+    @staticmethod
+    def _load_templates() -> list[MadlibsTemplate]:
+        madlibs_stories = Path("bot/resources/fun/madlibs_templates.json")
+
+        with open(madlibs_stories) as file:
+            return json.load(file)["templates"]
+
+    @staticmethod
+    def madlibs_embed(part_of_speech: str, number_of_inputs: int) -> discord.Embed:
+        """Method to have the bot send an embed with the game information."""
+        madlibs_embed = discord.Embed(
+            title="Madlibs",
+            color=Colours.python_blue,
+        )
+
+        madlibs_embed.add_field(
+            name="Enter a word that fits the given part of speech!",
+            value=f"Part of speech: {part_of_speech}"
+        )
+
+        madlibs_embed.set_footer(text=f"Inputs remaining: {number_of_inputs}")
+
+        return madlibs_embed
+
+    @commands.command()
+    async def madlibs(
+        self,
+        ctx: commands.Context,
+    ) -> None:
+        """
+        Play Madlibs with the bot!
+
+        Madlibs is a game where the player is asked to enter a word that
+        fits a random part of speech (e.g. noun, adjective, verb, plural noun, etc.)
+        a random amount of times, depending on the story chosen by the bot at the beginning.
+        """
+        random_template = choice(self.templates)
+
+        def author_check(message: discord.Message) -> bool:
+            return message.channel.id == ctx.channel.id and message.author.id == ctx.author.id
+
+        original_message = await ctx.send("Loading your Madlibs game...")
```

Comment:
I think the transition because it'll edit so fast it'll look a bit 'wonky' if that makes sense. Going from pure text to an embed might look that way.

You could set up an embed for this:
```suggestion
        
        loading_embed = discord.Embed(title="Madlibs", color=Colours.python_blue)
        original_message = await ctx.send(embed=loading_embed)
```

Reply:
@DMFriends this was marked resolved but I don't see the changes applied in the "Changed Files" tab.

Have you pushed your changes?

Reply:
Just pushed @Bluenix2